port/common/omrsock.c

```diff
@@ -0,0 +1,393 @@
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup Port
+ * @brief Sockets
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "omrport.h"
+#include "omrsock.h"
+
+/**
+ * Returns hints as a double pointer to an OMRAddInfoNode structure.
+ * 
+ * This hints structure is used to modify the results returned by a call to @ref omrsock_getaddrinfo. 
+ *
+ * @param[in] portLibrary The port library.
+ * @param[out] hints The filled-in hints structure
+ * @param[in] family A address family type
+ * @param[in] socktype A socket type
+ * @param[in] protocol A protocol family
+ * @param[in] flags Flags for modifying the result. If you wish to use multiple flags, perform the OR operation on the flags before passing it in.
+ *
+ * @return
+ * @arg 0, if no errors occurred, otherwise the (negative) error code
+ *
+ * @note current supported family types are:
+ * @arg OMRSOCK_AF_UNSPEC
+ * @arg OMRSOCK_AF_INET4
+ * @arg OMRSOCK_AF_INET6
+ *
+ * @note Works with both IPv4 and IPv6 support.
+ */
+int32_t
+omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t *hints, int family, int socktype, int protocol, int flags)
+{
+	return -1;
+}
+
+/**
+ * Answers a list of addresses as an opaque struct in "result".
+ * 
+ * Use the following functions to extract the details:
+ * @arg @ref omrsock_getaddrinfo_length
+ * @arg @ref omrsock_getaddrinfo_family
+ * @arg @ref omrsock_getaddrinfo_socktype
+ * @arg @ref omrsock_getaddrinfo_protocol
+
+ * @param[in] portLibrary The port library.
+ * @param[in] node The name of the host in either host name format or in IPv4 or IPv6 accepted notations.
+ * @param[in] service The port of the host in string form.
+ * @param[in] hints Hints on what results are returned (can be NULL for default action). Use omrsock_getaddrinfo_create_hints to create the hints.
+ * @param[out] result An opaque pointer to a list of results (OMRAddrInfoNode must be preallocated).
+ *
+ * @return	0, if no errors occurred, otherwise the (negative) error code.
```

Comment:
```suggestion
 * @return 0 if no errors occurred. Otherwise, <ADD_ERROR_CODE_NAME> in case of error.
```
inconsistent spacing